tools/runners/run-app.js

```diff
@@ -835,7 +839,13 @@ _.extend(AppRunner.prototype, {
 
     Console.enableProgressDisplay(false);
 
-    const postStartupResult = runPostStartupCallbacks(bundleResult);
+    const postStartupResult = Promise.race([
+      listenPromise,
+      new Promise(resolve => setTimeout(resolve, 3000))
```

Comment:
The `3000` used here seems a bit arbitrary/magic-y. Maybe this should be defined in clearly labelled constant.

Reply:
The commit message says "Also, the watcher and legacy bundle are only created after the server has started, or 3 seconds has passed." - it is necessary to have this 3 second time out at all? Why not always wait until the server starts?